#!/usr/bin/env python

PACKAGE = "dvs_displayer"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


select_display_method = gen.enum([gen.const('histogram',    str_t, 'histogram', 'Histogram'),
                                  gen.const('ternary',      str_t, 'ternary', 'Ternary'),
                                  gen.const('time_surface', str_t, 'time_surface', 'Time Surface')], 
                                  'display_method')

gen.add('display_method', str_t, 1, 'Display Method', 'ternary', edit_method=select_display_method)

select_color_map = gen.enum([gen.const('grayscale', str_t, 'grayscale', 'Grayscale'),
                             gen.const('seismic',   str_t, 'seismic', 'Seismic'),
                             gen.const('viridis',   str_t, 'viridis', 'Viridis')], 
                                  'color_map')

gen.add('color_map', str_t, 1, 'Color Map', 'seismic', edit_method=select_color_map)

gen.add("frequency", double_t, 1, "Frequency of the published image (-1 same as events)",    -1, -1,   100)

exit(gen.generate(PACKAGE, "displayer_node", "dvs_displayer"))
